spec/datadog/tracing/distributed/fetcher_spec.rb

```diff
@@ -2,69 +2,47 @@
 
 require 'spec_helper'
 
-require 'datadog/tracing/distributed/headers/parser'
+require 'datadog/tracing/distributed/fetcher'
 require 'datadog/tracing/span'
 
-RSpec.describe Datadog::Tracing::Distributed::Headers::Parser do
-  subject(:headers) do
-    described_class.new(env)
-  end
+RSpec.describe Datadog::Tracing::Distributed::Fetcher do
+  subject(:fetcher) { described_class.new(data) }
 
-  let(:env) { {} }
+  let(:data) { {} }
 
-  # Helper to format env header keys
-  def env_header(name)
-    "http-#{name}".upcase!.tr('-', '_')
-  end
+  describe '#[]' do
+    subject(:get) { fetcher[key] }
+    let(:key) {}
 
-  describe '#header' do
-    context 'header not present' do
-      it { expect(headers.header('request_id')).to be_nil }
+    context 'with no value associated' do
+      let(:key) { 'not present' }
+      it { is_expected.to be_nil }
     end
 
-    context 'header value is nil' do
-      let(:env) { { env_header('request_id') => nil } }
-
-      it { expect(headers.header('request_id')).to be_nil }
-    end
-
-    context 'header value is empty string' do
-      let(:env) { { env_header('request_id') => '' } }
-
-      it { expect(headers.header('request_id')).to be_nil }
-    end
-
-    context 'header is set' do
-      %w[
-        request_id
-        request-id
-        REQUEST_ID
-        REQUEST-ID
-        Request-ID
-      ].each do |header|
-        context "fetched as #{header}" do
-          let(:env) { { env_header('request_id') => 'rid' } }
-
-          it { expect(headers.header(header)).to eq('rid') }
-        end
-      end
+    context 'with a string value associated' do
+      let(:data) { { key => 'value' } }
+      it { is_expected.to eq('value') }
     end
```

Comment:
This test description is somewhat misleading, because it suggests that the fetcher is doing something based on a string value being associated or not, where actually that's not the case -- it works with any type.

```suggestion
    context 'with a value associated' do
      let(:data) { { key => 'value' } }
      it { is_expected.to eq('value') }
    end
```